src/index.js

```diff
@@ -0,0 +1,171 @@
+import {
+  title,
+  description,
+  ddate,
+  priority,
+  todoindex,
+  submit,
+  modal,
+  newTaskForm,
+  newProjectInput,
+  projects,
+  newProjectForm,
+  todoTemplate,
+  cards,
+  modalBtn,
+  closeModal,
+  modalHeader,
+} from './dom';
+
+import {
+  lists, save, object, clearList,
+} from './localStorage';
+import { AddTask, editTodo, addClass } from './addtodo';
+import createProject from './addproject';
+import './style.css';
+
+const deleteTask = (todo, card) => {
+  const editindex = todo.value;
+  const selectedProject1 = lists.find((list) => list.id === object.selectedId);
+  selectedProject1.tasks.splice(editindex, 1);
+
+  if (submit.classList.contains('edit')) {
+    submit.classList.remove('edit');
+  }
+  save();
+  cards.removeChild(card);
+};
+const renderTasks = (selectedId) => {
+  selectedId.tasks.forEach((task) => {
+    const todoList = document.importNode(todoTemplate.content, true);
+    const card = todoList.querySelector('.card');
+    const todoTitle = todoList.querySelector('.card-title');
+    todoTitle.innerText = task.title;
+    const todoDescription = todoList.querySelector('.card-description');
+    todoDescription.innerText = task.description;
+    const todoPriority = todoList.querySelector('.card-priority');
+    todoPriority.innerText = task.priority;
+    todoPriority.classList.add('btn');
+    const todoDdate = todoList.querySelector('.card-footer');
+    todoDdate.innerText = `Due on ${task.ddate}`;
+    const editBtn = todoList.querySelector('.edit-btn');
+    const index = selectedId.tasks.indexOf(task);
+
+    editBtn.addEventListener('click', () => editTodo(task, index));
+    const deleteTaskBtn = todoList.querySelector('.delete-btn');
+    deleteTaskBtn.addEventListener('click', () => {
+      deleteTask(todoindex, card);
+    });
+    cards.appendChild(todoList);
+    addClass();
+  });
+};
+
```

Comment:
this is pure DOM code

Reply:
- Moved most of the dom related functions to `dom.js` from other files.
- Added form validations, so you can not submit empty form fields.
- Changed button content on the add form or updating the form.

Reply:
We tried to move those to `dom.js`. Thanks for the review @shloch